scripts/lib/node_cache.py

```diff
@@ -90,7 +90,7 @@ def do_yarn_install(target_path, yarn_args, success_stamp, stdout=None, stderr=N
         if os.path.exists("node_modules"):
             cmds.append(["cp", "-R", "node_modules/", cached_node_modules])
         cd_exec = os.path.join(ZULIP_PATH, "scripts/lib/cd_exec")
-        cmds.append([cd_exec, target_path, YARN_BIN, "install", "--non-interactive"] +
+        cmds.append(['sudo', cd_exec, target_path, YARN_BIN, "install", "--non-interactive"] +
```

Comment:
`sudo` isn't something we want to run in here, because this tool needs to work in production systems where sudo may not be configured.  I added a linter check to prevent this.

Reply:
This is something that needs to be run as `sudo` in CircleCI, I will put this in if/else clause.

Reply:
I think it might be better to call the whole script using `sudo` in CircleCI?

Reply:
Ah! This script is imported by `tools/lib/provision.py` so we can run the whole script as `sudo`. So should i go with checking if `os.environ['CIRCLECI']`?

(And i am really too convinced that adding `sudo` is really the solution to this issue)